### PR TITLE
Remove user confirmation token

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This adds a confirmation token to the Quarterdeck database to support user remove confirmation, and updates the `UserDelete` endpoint (now `UserRemove`) to return a 200 response with the confirmation token, or a 204 if the user can just be removed without side effects.

Fixes SC-16120

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/quarterdeck/db/migrations/0005_user_org_logins.sql [hunk 1]
```diff
@@ -1,8 +1,11 @@
--- Add a column to the organization_users to record last login times.
+-- Add columns to the organization_users table.
 
 BEGIN;
 
 -- Last login time for the user in the organization.
 ALTER TABLE organization_users ADD COLUMN last_login TEXT DEFAULT NULL;
 
+-- Delete confirmation token for the organization user.
+ALTER TABLE organization_users ADD COLUMN delete_confirmation_token TEXT DEFAULT NULL;
```
Comment (pdeziel) [inline pkg/quarterdeck/db/migrations/0005_user_org_logins.sql:9]: Do we need to add this to a new migration since we've already deployed to staging?

Comment (bbengfort) [reply pkg/quarterdeck/db/migrations/0005_user_org_logins.sql:9]: We can delete the staging database; has this migration been applied to prod?

File: pkg/quarterdeck/users.go [hunk 2]
```diff
@@ -423,12 +429,27 @@ func (s *Server) UserDelete(c *gin.Context) {
...
+		out := &api.UserRemoveReply{
+			APIKeys: make([]string, 0, len(keys)),
+			Token:   token,
+		}
+
+		for _, key := range keys {
+			out.APIKeys = append(out.APIKeys, key.Name)
+		}
+
+		c.JSON(http.StatusOK, out)
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply pkg/quarterdeck/users.go:450]: 204 makes sense; alternatively we could put a `deleted` bool in the response struct and return it as true if the user was deleted and use 200 in both cases. I think @masskoder might prefer the latter.

**Review by bbengfort — APPROVED** [review]

File: pkg/quarterdeck/users.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pdeziel) [reply pkg/quarterdeck/users.go]: 200 makes sense to me

File: pkg/quarterdeck/db/migrations/0005_user_org_logins.sql (hunk 1 above)
Comment (pdeziel) [reply pkg/quarterdeck/db/migrations/0005_user_org_logins.sql:9]: I don't think it has because I just added the other migration a few days ago.

**Pull request merged**